keps/sig-node/4781-kublet-restart-pod-status/README.md

```diff
@@ -0,0 +1,705 @@
+<!-- /toc -->
+
+## Release Signoff Checklist
+
+Items marked with (R) are required *prior to targeting to a milestone / release*.
+
+- [ ] (R) Enhancement issue in release milestone, which links to KEP dir in [kubernetes/enhancements] (not the initial KEP PR)
+- [ ] (R) KEP approvers have approved the KEP status as `implementable`
+- [ ] (R) Design details are appropriately documented
+- [ ] (R) Test plan is in place, giving consideration to SIG Architecture and SIG Testing input (including test refactors)
+  - [ ] e2e Tests for all Beta API Operations (endpoints)
+  - [ ] (R) Ensure GA e2e tests meet requirements for [Conformance Tests](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/conformance-tests.md) 
+  - [ ] (R) Minimum Two Week Window for GA e2e tests to prove flake free
+- [ ] (R) Graduation criteria is in place
+  - [ ] (R) [all GA Endpoints](https://github.com/kubernetes/community/pull/1806) must be hit by [Conformance Tests](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/conformance-tests.md) 
+- [ ] (R) Production readiness review completed
+- [ ] (R) Production readiness review approved
+- [ ] "Implementation History" section is up-to-date for milestone
+- [ ] User-facing documentation has been created in [kubernetes/website], for publication to [kubernetes.io]
+- [ ] Supporting documentation—e.g., additional design documents, links to mailing list discussions/SIG meetings, relevant PRs/issues, release notes
+
+<!--
+**Note:** This checklist is iterative and should be reviewed and updated every time this enhancement is being considered for a milestone.
+-->
+
+[kubernetes.io]: https://kubernetes.io/
+[kubernetes/enhancements]: https://git.k8s.io/enhancements
+[kubernetes/kubernetes]: https://git.k8s.io/kubernetes
+[kubernetes/website]: https://git.k8s.io/website
+
+## Summary
+
+When a kubelet restarts, pods with configured readiness probes may momentarily report containerNotReady status, potentially leading to service disruptions. This KEP proposes enhancing kubelet's pod readiness management to ensure that pod readiness status is preserved across kubelet restarts.
+
+## Motivation
+
+Maintaining high availability and minimizing service interruptions is crucial for Kubernetes clusters. When the kubelet restarts, it re-evaluates the state of all containers and pods. This can sometimes lead to a situation where the container's Ready state, which was previously set based on the success of its readiness probes, gets reset. This reset can cause services to be incorrectly marked as unavailable, even though they were functioning correctly before the kubelet restart.
+
+### Goals
+
+- Maintain the previous ready state of containers after kubelet restart until a new probe result is available.
+- Ensure consistency in container ready states across kubelet restarts.
+- Minimize unnecessary service disruptions caused by temporary ready state changes.
```

Comment:
Can this also apply to a pod with startupProbe or can this KEP cover that? https://github.com/kubernetes/kubernetes/issues/100277#issuecomment-1563686325

Reply:
+1 startupProbe too should be considered

Reply:
Thank you for your suggestion. We plan to inherit the Start state from the previous container if it starts before Kubelet, instead of waiting for the startupProbe. The specific details will be included in the new commit.